mithril-client/tests/extensions/snapshot_archives.rs

```diff
@@ -70,23 +77,35 @@ pub fn build_immutable_files_archives(cardano_db: &DummyCardanoDb, target_dir: &
         let enc = zstd::Encoder::new(tar_file, 3).unwrap();
         let mut tar = tar::Builder::new(enc);
 
-        for extension in &[".chunk", ".primary", ".secondary"] {
-            let file_name = format!("{:05}{}", immutable_number, extension);
-            let file_path = immutable_dir.join(&file_name);
-
-            let archive_path = format!("immutable/{file_name}");
-            tar.append_path_with_name(&file_path, &archive_path)
-                .unwrap();
-        }
+        append_immutable_trio(&mut tar, immutable_number, immutable_dir);
 
         let zstd = tar.into_inner().unwrap();
         zstd.finish().unwrap();
     }
 }
 
-pub fn build_ancillary_files_archive(cardano_db: &DummyCardanoDb, target_dir: &Path) {
+fn append_immutable_trio<T: std::io::Write>(
+    tar: &mut tar::Builder<T>,
+    immutable_file_number: ImmutableFileNumber,
+    immutables_dir: &Path,
+) {
+    for extension in &[".chunk", ".primary", ".secondary"] {
```

Comment:
We can extract a function to build trio file_paths and then, we can reuse it in `build_ancillary_files_archive` to avoid multiple construction of this file_path.